hpx/parallel/algorithms/transform.hpp

```diff
@@ -179,6 +180,37 @@ namespace hpx { namespace parallel { inline namespace v1
                     >::get(std::make_pair(std::move(first), std::move(dest)));
             }
         };
+
+        /// non_segmented version
+        template <typename ExPolicy, typename InIter, typename OutIter, typename F,
+            typename Proj>
+        typename util::detail::algorithm_result<ExPolicy,
+            hpx::util::tagged_pair<tag::in(InIter), tag::out(OutIter)>
+        >::type
+        transform_(ExPolicy && policy, InIter first, InIter last, OutIter dest,
+            F && f, Proj && proj, std::false_type)
+        {
+            typedef std::integral_constant<bool,
+                    parallel::execution::is_sequenced_execution_policy<ExPolicy>
+                    ::value || !hpx::traits::is_forward_iterator<InIter>::value ||
+                   !hpx::traits::is_forward_iterator<OutIter>::value
```

Comment:
The Boolean expression is very difficult to read thus way. Could you please try to keep every term separate, if possible, e.g.:

    parallel::execution::is_sequenced_execution_policy<ExPolicy>::value || 
    !hpx::traits::is_forward_iterator<InIter>::value || 
    !hpx::traits::is_forward_iterator<OutIter>::value